### PR TITLE
Alerting: Fix UI toggles and search input synchronization

### PR DESCRIPTION
**Pull request by konrad147**
Fix for broken UI toggles and search input synchronization.

**Special notes for your reviewer**:
Caused by the different handling of search input

### CODE REVIEW

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM!

File: public/app/features/alerting/unified/components/rules/RulesFilter.test.tsx [hunk 1]
```diff
@@ -13,23 +18,68 @@ jest.mock('@grafana/runtime', () => {
...
+  },
+  searchInput: byLabelText('Search'),
+};
+
+beforeEach(() => {
+  locationService.replace({ search: '' });
+});
+
+describe('RulesFilter', () => {
+  it('Should apply state filter to the search input', async () => {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/RulesFilter.test.tsx:45]: 👏

**Review by VikaCep — APPROVED** [review]
👏 👏 👏

**Review by gillesdemey — APPROVED** [review]
LGTM after linting issues have been resolved ✅

**Pull request merged**